src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/controllers/MonitorTaskJobControllerForTerminationJobTest.java

```diff
@@ -0,0 +1,101 @@
+import uk.gov.hmcts.reform.wataskmonitor.domain.taskmonitor.JobName;
+import uk.gov.hmcts.reform.wataskmonitor.domain.taskmonitor.request.JobDetails;
+import uk.gov.hmcts.reform.wataskmonitor.domain.taskmonitor.request.MonitorTaskJobRequest;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.wacaseeventhandler.controllers.MonitorTaskJobControllerUtility.expectedResponse;
+import static uk.gov.hmcts.reform.wataskmonitor.domain.taskmanagement.request.enums.TerminateReason.CANCELLED;
+import static uk.gov.hmcts.reform.wataskmonitor.domain.taskmanagement.request.enums.TerminateReason.COMPLETED;
+
+class MonitorTaskJobControllerForTerminationJobTest extends SpringBootIntegrationBaseTest {
+
+    public static final String SERVICE_TOKEN = "some service token";
+    public static final String CAMUNDA_TASK_ID_FOR_CANCELLATION = "some camunda task id for cancellation";
+    public static final String CAMUNDA_TASK_ID_FOR_COMPLETION = "some camunda task id for completion";
+
+    @MockBean
+    private CamundaClient camundaClient;
+    @MockBean
+    private AuthTokenGenerator authTokenGenerator;
+    @MockBean
+    private TaskManagementClient taskManagementClient;
+
+    @BeforeEach
+    void setUp() {
+        mockExternalDependencies();
+    }
+
+    @SuppressWarnings({"PMD.JUnitTestsShouldIncludeAssert", "PMD.LawOfDemeter"})
+    @Test
+    public void shouldSucceedAndTerminateTasks() throws Exception {
+        MonitorTaskJobRequest monitorTaskJobReq = new MonitorTaskJobRequest(new JobDetails(JobName.TERMINATION));
+
+        mockMvc.perform(post("/monitor/tasks/jobs")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(TestUtility.asJsonString(monitorTaskJobReq)))
+            .andExpect(status().isOk())
+            .andExpect(content().string(equalTo(expectedResponse.apply(JobName.TERMINATION.name()))));
+
+        verify(authTokenGenerator).generate();
+        verify(camundaClient).getTasksFromHistory(
+            eq(SERVICE_TOKEN),
+            eq("0"),
+            eq("1000"),
+            any()
+        );
+
+        verifyTerminateEndpointWasCalledWithTerminateReason(CANCELLED, 1, CAMUNDA_TASK_ID_FOR_CANCELLATION);
+        verifyTerminateEndpointWasCalledWithTerminateReason(COMPLETED, 1, CAMUNDA_TASK_ID_FOR_COMPLETION);
+    }
+
```

Comment:
Tests  to cover AC's 2 & 3?
1. When no task is returned, then complete successfully and return empty list
2. No response or timeout from Camunda, then exception should be thrown